### PR TITLE
[ci] improvments to make pipeline faster

### PR DESCRIPTION
**Pull request by alvicsam**
Nextest has a nice ability to [partition or shard tests into buckets](https://nexte.st/book/partitioning.html)
Gitlab can [run one job in parallel on several runners](https://docs.gitlab.com/ee/ci/yaml/#parallel)

Combining this two approaches and applying them to the `test-linux-stable` job allows to save time. 
It is also required to remove `continuous-integration/gitlab-test-linux-stable` from Required jobs and add
 - `continuous-integration/gitlab-test-linux-stable 1/3`
 - `continuous-integration/gitlab-test-linux-stable 2/3`
 - `continuous-integration/gitlab-test-linux-stable 3/3`

And the same for `continuous-integration/gitlab-cargo-check-benches`:
 - continuous-integration/gitlab-cargo-check-benches 1/2
 - continuous-integration/gitlab-cargo-check-benches 2/2

Added nextest to `test-wasmer-sandbox` and applied the approach which I described above.
Added `parallel: 2` to execute benchmarks on 2 different runners at the same time

Closes https://github.com/paritytech/ci_cd/issues/207

### CODE REVIEW

**Comment by alvicsam** [comment]
bot rebase

File: scripts/ci/gitlab/pipeline/test.yml [hunk 1]
```diff
@@ -213,14 +214,26 @@ test-linux-stable:
...
     RUN_UI_TESTS:                  1
+    # needed for rusty-cachier to keep cache in test-linux-stable folder and not in test-linux-stable-1/3
+    CI_JOB_NAME:                   "test-linux-stable"
+  parallel: 3
   script:
     - rusty-cachier snapshot create
-    # TODO: add to paritytech/ci-linux image
+    # TODO: remove when current paritytech/ci-linux:staging (with rust stable 1.62) is switched to production
     - time cargo install cargo-nextest
     # this job runs all tests in former runtime-benchmarks, frame-staking and wasmtime tests
```
Comment (TriplEight) [inline scripts/ci/gitlab/pipeline/test.yml:224]:
```suggestion
    # this job runs all tests in former runtime-benchmarks, frame-staking and wasmtime tests
    # tests are partitioned by nextest and executed in parallel on $CI_NODE_TOTAL runners
```

File: scripts/ci/gitlab/pipeline/test.yml [hunk 2]
```diff
@@ -74,7 +74,8 @@ cargo-check-nixos:
 cargo-check-benches:
   stage:                           test
   variables:
-    RUSTY_CACHIER_TOOLCHAIN: nightly
+    RUSTY_CACHIER_TOOLCHAIN:       nightly
+    CI_JOB_NAME:                   cargo-check-benches
```
[the remark this answers is not in the document]
Comment (alvicsam) [reply scripts/ci/gitlab/pipeline/test.yml:78]: >what is it used for?

need help from @rcny

File: scripts/ci/gitlab/pipeline/test.yml [hunk 3]
```diff
@@ -91,16 +92,16 @@ cargo-check-benches:
...
     - rusty-cachier snapshot create
     - mkdir -p ./artifacts/benches/$CI_COMMIT_REF_NAME-$CI_COMMIT_SHORT_SHA
-    - SKIP_WASM_BUILD=1 time cargo +nightly check --benches --all
-    - 'cargo run --release -p node-bench -- ::node::import::native::sr25519::transfer_keep_alive::paritydb::small --json
-      | tee ./artifacts/benches/$CI_COMMIT_REF_NAME-$CI_COMMIT_SHORT_SHA/::node::import::native::sr25519::transfer_keep_alive::paritydb::small.json'
-    - 'cargo run --release -p node-bench -- ::trie::read::small --json
-      | tee ./artifacts/benches/$CI_COMMIT_REF_NAME-$CI_COMMIT_SHORT_SHA/::trie::read::small.json'
-    - sccache -s
-    - rusty-cachier cache upload
+    - if [ ${CI_NODE_INDEX} == 1 ]; then SKIP_WASM_BUILD=1 time cargo +nightly check --benches --all; fi
```
[the remark this answers is not in the document]
Comment (alvicsam) [reply scripts/ci/gitlab/pipeline/test.yml]: fixed

File: scripts/ci/gitlab/pipeline/test.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (alvicsam) [reply scripts/ci/gitlab/pipeline/test.yml]: added comment and quotes

File: scripts/ci/gitlab/pipeline/test.yml [hunk 4]
```diff
@@ -97,12 +98,21 @@ cargo-check-benches:
...
+            | tee ./artifacts/benches/$CI_COMMIT_REF_NAME-$CI_COMMIT_SHORT_SHA/::trie::read::small.json;
+          echo "___Uploading cache for rusty-cachier___";
+          rusty-cachier cache upload
+          ;;
+        2)
+          echo "___Running benchmarks___";
+          cargo run --release -p node-bench -- ::node::import::native::sr25519::transfer_keep_alive::paritydb::small --json
+            | tee ./artifacts/benches/$CI_COMMIT_REF_NAME-$CI_COMMIT_SHORT_SHA/::node::import::native::sr25519::transfer_keep_alive::paritydb::small.json
+          ;;
+        esac
```
Comment (TriplEight) [inline scripts/ci/gitlab/pipeline/test.yml:115]:
```suggestion
    - echo "___Running benchmarks___";
    - case ${CI_NODE_INDEX} in
        1)
          SKIP_WASM_BUILD=1 time cargo +nightly check --benches --all;
          cargo run --release -p node-bench -- ::trie::read::small --json
            | tee ./artifacts/benches/$CI_COMMIT_REF_NAME-$CI_COMMIT_SHORT_SHA/::trie::read::small.json;
          echo "___Uploading cache for rusty-cachier___";
          rusty-cachier cache upload
          ;;
        2)
          cargo run --release -p node-bench -- ::node::import::native::sr25519::transfer_keep_alive::paritydb::small --json
            | tee ./artifacts/benches/$CI_COMMIT_REF_NAME-$CI_COMMIT_SHORT_SHA/::node::import::native::sr25519::transfer_keep_alive::paritydb::small.json
          ;;
        esac
```

**Review by TriplEight — CHANGES_REQUESTED** [review]

**Review by radupopa2010 — APPROVED** [review]

**Comment by alvicsam** [comment]
@nazar-pc thank you for your suggestion. It'll look nicer but it might be not obvious where details could be found. So now each job has a `details` link which leads to logs for this particular job. With a cumulative check the only thing which I can think of is to put a link to the whole pipeline in `details`. But it might be not clear why this check has failed.

**Review by TriplEight — APPROVED** [review]

**Pull request merged**